cpp/modmesh/transform/fourier.hpp

```diff
@@ -87,6 +91,67 @@ void ifft(SimpleArray<T1<T2>> const & in, SimpleArray<T1<T2>> & out)
     }
 }
 
+template <template <typename> class T1, typename T2>
+void fft_bluestein(SimpleArray<T1<T2>> const & in, SimpleArray<T1<T2>> & out)
+{
+    const size_t N = in.size();
+    // Calculate a length with power of 2 and at least 2N-1
+    const size_t K = detail::next_power_of_two(2 * N - 1);
+
+    SimpleArray<T1<T2>> a{modmesh::small_vector<size_t>{K}, T1<T2>{0.0, 0.0}};
+    SimpleArray<T1<T2>> A{modmesh::small_vector<size_t>{K}, T1<T2>{0.0, 0.0}};
+    SimpleArray<T1<T2>> b{modmesh::small_vector<size_t>{K}, T1<T2>{0.0, 0.0}};
+    SimpleArray<T1<T2>> B{modmesh::small_vector<size_t>{K}, T1<T2>{0.0, 0.0}};
+
+    // Calculate a[0], b[0] first, becuase it can avoid a branch in the following
+    // for loop!
+    a[0] = in[0];
+    b[0] = T1<T2>{1.0, 0.0};
+
+    for (size_t i = 1; i < N; ++i)
+    {
+        T2 tmp = -pi<T2> * i * i / static_cast<T2>(N);
+        T1<T2> twiddle_factor{std::cos(tmp), std::sin(tmp)};
+
+        a[i] = in[i] * twiddle_factor;
+        b[i] = twiddle_factor.conj();
+        // Convert circular convolution to linear convolution
+        b[K - i] = b[i];
+    }
+
+    fft_radix_2<T1, T2>(a, A);
+    fft_radix_2<T1, T2>(b, B);
+
+    for (size_t i = 0; i < K; ++i)
+    {
+        A[i] *= B[i];
+    }
+
+    ifft<T1, T2>(A, a);
+
+    for (size_t i = 0; i < N; ++i)
+    {
+        T2 tmp = -pi<T2> * i * i / static_cast<T2>(N);
+        T1<T2> twiddle_factor{std::cos(tmp), std::sin(tmp)};
+        out[i] = a[i] * twiddle_factor;
+    }
+}
+
+template <template <typename> class T1, typename T2>
+void fft(SimpleArray<T1<T2>> const & in, SimpleArray<T1<T2>> & out)
+{
+    const size_t N = in.size();
+
+    if ((N & (N - 1)) == 0)
+    {
+        fft_radix_2<T1, T2>(in, out);
+    }
+    else
+    {
+        fft_bluestein<T1, T2>(in, out);
+    }
+}
```

Comment:
If the input array length is power of 2 (radix-2) then using radix-2 fft otherwise using Bluestein fft.